crates/bevy_render/src/camera/camera.rs

```diff
@@ -1,46 +1,115 @@
-use std::marker::PhantomData;
-
 use crate::{
     camera::CameraProjection,
     prelude::Image,
     render_asset::RenderAssets,
     render_resource::TextureView,
     view::{ExtractedView, ExtractedWindows, VisibleEntities},
-    RenderApp, RenderStage,
 };
-use bevy_app::{App, CoreStage, Plugin, StartupStage};
 use bevy_asset::{AssetEvent, Assets, Handle};
+use bevy_derive::{Deref, DerefMut};
 use bevy_ecs::{
     change_detection::DetectChanges,
     component::Component,
     entity::Entity,
     event::EventReader,
-    prelude::With,
     query::Added,
     reflect::ReflectComponent,
-    system::{Commands, ParamSet, Query, Res, ResMut},
+    system::{Commands, ParamSet, Query, Res},
 };
 use bevy_math::{Mat4, UVec2, Vec2, Vec3};
 use bevy_reflect::prelude::*;
 use bevy_transform::components::GlobalTransform;
 use bevy_utils::HashSet;
 use bevy_window::{WindowCreated, WindowId, WindowResized, Windows};
 use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
 use wgpu::Extent3d;
 
-#[derive(Component, Default, Debug, Reflect, Clone)]
-#[reflect(Component, Default)]
+#[derive(Component, Debug, Reflect, Clone)]
+#[reflect(Component)]
 pub struct Camera {
     pub projection_matrix: Mat4,
+    pub logical_target_size: Option<Vec2>,
+    pub physical_target_size: Option<UVec2>,
+    pub priority: isize,
+    pub is_active: bool,
     #[reflect(ignore)]
     pub target: RenderTarget,
     #[reflect(ignore)]
     pub depth_calculation: DepthCalculation,
-    pub near: f32,
-    pub far: f32,
 }
 
-#[derive(Debug, Clone, Reflect, PartialEq, Eq, Hash)]
+impl Default for Camera {
+    fn default() -> Self {
+        Self {
+            is_active: true,
+            priority: 0,
+            projection_matrix: Default::default(),
+            logical_target_size: Default::default(),
+            physical_target_size: Default::default(),
+            target: Default::default(),
+            depth_calculation: Default::default(),
+        }
+    }
+}
+
+impl Camera {
+    /// Given a position in world space, use the camera to compute the screen space coordinates.
+    ///
+    /// To get the coordinates in Normalized Device Coordinates, you should use
+    /// [`world_to_ndc`](Self::world_to_ndc).
+    pub fn world_to_screen(
+        &self,
+        camera_transform: &GlobalTransform,
+        world_position: Vec3,
+    ) -> Option<Vec2> {
+        let window_size = self.logical_target_size?;
```

Comment:
Is `screen` the best term here ? As it's not tied to the window anymore, is "view" a better alternative ? The method name becoming `world_to_view`

Reply:
`view` in terms of coordinates is used for points transformed by the model and view transforms. I agree `window_size` may be a misnomer, but in that case I think `target_size` makes more sense.

Reply:
Yeah this should be `target_size`.

I agree that `world_to_screen` is no longer fully accurate (and it will get even less accurate when we add viewports). Godot calls this operation on their Cameras `unproject_position` (within the context of their high level Viewports). Unity has separate WorldToScreenPoint and WorldToViewportPoint. Unreal calls this World to Screen, with a flag to make it "relative to the viewport".

I think `world_to_viewport` is what we should call this method. Then we can add `world_to_target` (or alternatively `world_to_screen`) once we add viewports and the distinction is useful.

Reply:
I renamed this to world_to_viewport, but feel free to suggest alternatives.